File: 137-single-number-ii/single-number-ii.java
class Solution {
    public int singleNumber(int[] nums) {
        int result = 0;
        for(int bitposition = 0; bitposition < 32; bitposition++){
            int bitcount = 0;

            // for how many nos. we have 1 at the currnet bit posiion
            for(int num: nums){
                bitcount += (num >> bitposition) & 1;
            }
            bitcount %= 3;
            result |= (bitcount << bitposition);
        }
        return result;
    }
}

File: 137-single-number-ii/single-number-ii.go
func singleNumber(nums []int) int {
    result := 0
    for bitposition := 0; bitposition < 32; bitposition++ {
        bitcount := 0
        for _, num := range nums {
            bitcount += (num >> bitposition) & 1
        }
        if bitcount % 3 != 0 {
            result |= (1 << bitposition)
        }
    }
    return int(int32(result))
}